iothub/service/src/Twin/Models/ClientTwinMetadata.cs

```diff
@@ -10,28 +13,49 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public sealed class ClientTwinMetadata
     {
-        /// <summary>
-        /// Initializes an instance of this class.
-        /// </summary>
-        /// <param name="lastUpdatedOnUtc">When a property was last updated.</param>
-        /// <param name="lastUpdatedVersion">The version of the property when last updated.</param>
-        public ClientTwinMetadata(DateTimeOffset lastUpdatedOnUtc, long? lastUpdatedVersion)
-        {
-            LastUpdatedOnUtc = lastUpdatedOnUtc;
-            LastUpdatedVersion = lastUpdatedVersion;
-        }
-
         /// <summary>
         /// When a property was last updated.
         /// </summary>
-        public DateTimeOffset LastUpdatedOnUtc { get; set; }
+        [JsonProperty("$lastUpdated", DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public DateTimeOffset? LastUpdatedOnUtc { get; set; }
 
         /// <summary>
         /// The version of the property when last updated.
         /// </summary>
         /// <remarks>
-        /// This should be null for reported properties metadata and must not be null for desired properties metadata.
+        /// This should be not be included for reported properties metadata and must not be null for desired properties metadata.
         /// </remarks>
+        [JsonProperty("$lastUpdatedVersion", DefaultValueHandling = DefaultValueHandling.Ignore)]
         public long? LastUpdatedVersion { get; set; }
+
+        [JsonExtensionData]
+        internal IDictionary<string, JToken> Properties { get; set; } = new Dictionary<string, JToken>();
+
+        /// <summary>
+        /// Gets the specified property's metadata by name as another metadata object.
+        /// </summary>
+        /// <param name="propertyName">The name of the property.</param>
+        /// <param name="propertyValue">The property's metadata.</param>
+        /// <returns>True if the property exists and could be converted, otherwise false.</returns>
+        public bool TryGetPropertyMetadata(string propertyName, out ClientTwinMetadata propertyValue)
```

Comment:
This should only ever be ClientTwinMetadata, so we could consider changing to `GetPropertyMetadata` which returns the property, but I figured we might as well follow the same pattern as elsewhere in the Twin code.

Thoughts?

Reply:
I like TryGet here, for the reason you mentioned about maintaining the same pattern

Reply:
For nested metadata, TryGet would be useful, for example - someone tries to get a metadata for a property update that doesn't exist

Reply:
> For nested metadata, TryGet would be useful, for example - someone tries to get a metadata for a property update that doesn't exist

True!